relayer/pkg/chainlink/ocr2/medianreport/report.go

```diff
@@ -59,7 +64,11 @@ func (c ReportCodec) BuildReport(oo []median.ParsedAttributedObservation) (types
 	var observations []junotypes.Felt
 	for i, o := range oo {
 		observers[i] = byte(o.Observer)
-		observations = append(observations, junotypes.BigToFelt(o.Value))
+		obsBytes, err := bigbigendian.SerializeSigned(junotypes.FeltLength, o.Value)
+		if err != nil {
+			return nil, errors.Wrap(err, "error in serializing observation")
+		}
+		observations = append(observations, junotypes.BytesToFelt(obsBytes))
```

Comment:
I think you just need to use the same functions I used for transmission: `parseAnswer` and `signedFelt` for parsing: https://github.com/smartcontractkit/chainlink-starknet/blob/7b9676a8e87b5f474c015ad0e4e8b7b788e950d4/relayer/pkg/chainlink/ocr2/utils.go#L75-L98 this `mod` for serializing: https://github.com/smartcontractkit/chainlink-starknet/blob/7b9676a8e87b5f474c015ad0e4e8b7b788e950d4/relayer/pkg/chainlink/ocr2/config_digester.go#L69-L70

Cairo has no notion of signed values: negative values have to be wrapped into the upper half of `PRIME` (so `0` to `PRIME/2` is positive, `PRIME/2` to `PRIME` is negative) or transmission will fail. Try removing this function in aggregator tests for example:

https://github.com/smartcontractkit/chainlink-starknet/blob/7b9676a8e87b5f474c015ad0e4e8b7b788e950d4/contracts/test/ocr2/aggregator-test.ts#L14-L18